### PR TITLE
chore: switch from JSHint to ESLint

### PR DESCRIPTION
**Pull request by ariporad**
This PR switches us from JSHint to ESLint.

ESLint checks both lint errors and style.

We're using [AirBnB's Style Guide](https://github.com/airbnb/javascript) as a starting point.

Since there are about 1700 lint errors, which I couldn't possibly hope to do myself, this PR is going to be done slightly differently. While this PR will remain open, Me and @nfischer will accept PRs against ***this*** branch fixing a small number of lint errors (Please do one file per PR). Then, once the number of lint errors is 0, we'll squash everything and merge into master.

Current Lint Problem Count for this branch: `794`

### CODE REVIEW

**Comment by nfischer** [comment]
@ariporad is this up to date with master? It looks like it's failing Windows CI.

Also, it looks like something is undefined in `run-tests.js`

**Comment by ariporad** [comment]
@nfischer: It's up to date with master, but the lint stage is failing ***badly***

**Comment by ariporad** [comment]
@nfischer: Oh, actually, `spawnSync` isn't available on node v0.10. Fixed.

**Comment by nfischer** [comment]
@ariporad I changed the milestone, since I don't think we should push off 0.6 just so we can switch lint tools

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (BYK) [inline .eslintrc]: We should prefer `.eslintrc.json` as the file name. "raw" `.eslintrc` files are deprecated.

**Comment by BYK** [comment]
I'm willing to help with this. Have you tried using the `--fix` option to get some of these fixed autoamtically?

File: .eslintrc (hunk 1 above)
Comment (ariporad) [inline .eslintrc]: Good catch

**Comment by ariporad** [comment]
@BYK: Oops... I forgot about that.

**Comment by ariporad** [comment]
@BYK: Fixed.

**Comment by BYK** [comment]
Adding the following to the rule set helps a lot:

    "spaced-comment": [2, "always", { "markers": ["@"], "exceptions": ["@"] }]

**Comment by BYK** [comment]
Looks like this is the problem:

    /home/travis/build/shelljs/shelljs/test/which.js
    18:5  error  "result" is already defined  no-redeclare

**Comment by BYK** [comment]
FYI I didn't try to fix all warnings so the warnings are expected. `no-param-reassign` is extensively violated and requires some fundamental changes in a few places so I didn't bother fixing them.

Also I'm not sure AirBnb style guide was a good choice since it looks a bit too browser oriented.

**Comment by BYK** [comment]
>(Please do one file per PR)

lol - sorry for not following this :D

**Comment by ariporad** [comment]
Don't worry, it didn't end up making a difference.

**Comment by TimothyGu** [comment]
Umm… is there a reason why this is high priority?

**Comment by nfischer** [comment]
I feel like this should be a low priority issue. Lengthy chores probably shouldn't be high priority. I think only in-demand features and bug fixes (and issues blocking those) should be high priority

**Comment by ariporad** [comment]
Oh, yeah, that was a mistake. Fixed.

**Comment by nfischer** [comment]
@ariporad any further work on this? This might be nice to start transitioning to using.

**Pull request closed without merging**